src/components/Code/Code.js

```diff
@@ -17,27 +21,56 @@ const captionStyle = css`
   border-bottom: none;
 `;
 
+const sourceCodeStyle = css`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
 const getLanguage = (lang) => {
   if (Object.values(Language).includes(lang)) {
     return lang;
   } else if (lang === 'sh') {
     // Writers commonly use 'sh' to represent shell scripts, but LeafyGreen and Highlight.js use the key 'shell'
     return 'shell';
-  } else if (['c', 'cpp', 'csharp'].includes(lang)) {
-    // LeafyGreen renders all C-family languages with "clike"
-    return 'clike';
+  } else if (['c', 'cpp'].includes(lang)) {
+    // LeafyGreen renders C and C++ languages with "cs"
+    return 'cs';
   }
   return 'none';
 };
 
-const Code = ({ nodeData: { caption, copyable, emphasize_lines: emphasizeLines, lang, linenos, value } }) => {
+const Code = ({ nodeData: { caption, copyable, emphasize_lines: emphasizeLines, lang, linenos, value, source } }) => {
   const { setActiveTab } = useContext(TabContext);
   const { languageOptions, codeBlockLanguage } = useContext(CodeContext);
   const code = value;
   const language = (languageOptions?.length > 0 && codeBlockLanguage) || getLanguage(lang);
   const captionSpecified = !!caption;
+  const sourceSpecified = !!source;
   const captionBorderRadius = captionSpecified ? '0px' : '4px';
 
+  let customActionButtonList = [];
+  if (sourceSpecified) {
+    customActionButtonList = [
+      <IconButton href={source}>
+        <Tooltip
+          triggerEvent="hover"
+          align="bottom"
+          justify="middle"
+          trigger={
+            <div css={sourceCodeStyle}>
+              <Icon glyph="OpenNewTab" />
+            </div>
+          }
+          darkMode={true}
+          popoverZIndex={2}
+        >
+          View full source
+        </Tooltip>
+      </IconButton>,
```

Comment:
Do you know if something like this works? `IconButton` should wrap its children with a `div` that applies the same centering CSS as `sourceCodeStyle`. If it doesn't work, I'm down to keep it as is!

```suggestion
      <Tooltip
        triggerEvent="hover"
        align="bottom"
        justify="middle"
        trigger={
          <IconButton href={source}>
            <Icon glyph="OpenNewTab" />
          </IconButton>
        }
        darkMode={true}
        popoverZIndex={2}
      >
        View full source
      </Tooltip>,
```

Reply:
I tried this and it doesn't seem to work - think it might be better to keep it as is!